lib/tasks/accounts/update_adopter_status.rake

```diff
@@ -0,0 +1,36 @@
+namespace :accounts do
+  desc 'Update newly created field for adopter status from Salesforce'
+  # rake accounts:update_adopter_status
+  task update_adopter_status: [:environment] do
+    last_id = nil
+    loop do
+      users = User.where(adopter_status: nil).order(:id).limit(250)
+      users = users.where("id > #{last_id}") unless last_id.nil?
+      last_id = users.last&.id
+
+      begin
+        contacts = OpenStax::Salesforce::Remote::Contact.select(
+          :id,
+          :adoption_status,
+          :accounts_uuid
+        )
+        .where(id: users.map(&:salesforce_contact_id))
+        .index_by(&:id)
```

Comment:
I'm a little fuzzy on if I did this correctly, will this work properly with line 22 to map the correct contact to the user?

Reply:
You need to remove `.to_a` in line 19 below if you want a hashmap.